### ISSUE TITLE
Cannot specify custom Python shebang for hook executable

### ISSUE THREAD

**Issue by kbsezginel** [pre-commit/pre-commit#1312]
I am working on a shared Python environment with `python3.6` as executable and I want the shebang to point to the correct executable `#!/usr/bin/env python3.6`. However, even though I specify 
```
default_language_version:
    python: python3.6
```
or  `language_version: python3.6` in `hooks`, it always picks `#!/usr/bin/env python` as the shebang even though there is no `python` executable. Is there a way to specify a custom shebang?

**Comment by asottile** [comment]
could you provide a reproduction? I'm not sure which shebang you are referring to

**Comment by kbsezginel** [comment]
Sure, I am using the following `.pre-commit-config.yaml`:
```
default_language_version:
    python: python3.6
repos:
  - repo: https://github.com/psf/black
    rev: stable
    hooks:
        - id: black
```
and when I run `pre-commit install` it adds `#!/usr/bin/env python` to the hook executable (`.git/hooks/pre-commit`). 

I am on `pre-commit 2.0.1`.

**Comment by asottile** [comment]
pre-commit tries to detect pythons that are reliably installed and falls back to `python` if none exist: https://github.com/pre-commit/pre-commit/blob/78f406af8a7099e74ebf458818633664cb5404d8/pre_commit/commands/install_uninstall.py#L52-L68

meaning you probably don't have a `python3.6` installed in `os.defpath`

**Comment by asottile** [comment]
(the configuration setting doesn't affect that at all, the configuration setting is for how hook environments are configured/installed)

**Comment by kbsezginel** [comment]
I see, it turns out that `python3.6` is actually installed in `/usr/local/bin/python3.6`.

**Comment by kbsezginel** [comment]
It still works when I change the shebang to `#!/usr/bin/env python3.6` though.

**Comment by asottile** [comment]
what is your value of `os.defpath` and what does `which -a python` and `which -a python3.6` show

**Comment by kbsezginel** [comment]
- `os.defpath` is `':/bin:/usr/bin'`
- `which python` returns `python: Command not found.` as expected
- `which python3.6` returns `/usr/local/bin/python3.6`
I am not sure about the `-a` option, it's not recognized. It is a FreeBSD shell if that makes any difference.

**Comment by asottile** [comment]
interesting :thinking: -- I'll have to think about this some more -- you might be able to work around with `ln -sf $(which python3.6) ~/bin/python` for now at least

my only thought that would "fix" this would be to add `/usr/local/bin` to the places we look always ?

**Comment by asottile** [comment]
alright I'm going to do a couple of things here:

- also look at `/usr/local/bin`
- failing to match: use `basename(readlink(sys.executable))`

this ~should result in more reliable matches on posix platforms

**Comment by asottile** [comment]
this has been released as part of [v2.1.0](https://github.com/pre-commit/pre-commit/releases/tag/v2.1.0) -- thanks again for the report!

**Comment by kbsezginel** [comment]
Great, thank you for the quick replies!
